packages/host/tests/integration/components/card-field-separation-test.gts

```diff
@@ -0,0 +1,202 @@
+      .loader;
+    shimExternals(loader);
+    cardApi = await loader.import(`${baseRealm.url}card-api`);
+    string = await loader.import(`${baseRealm.url}string`);
+    number = await loader.import(`${baseRealm.url}number`);
+
+    createTestRealm = async (files) => {
+      let realm = await TestRealm.create(loader, files, this.owner);
+      await realm.ready;
+    };
+
+    setCardInOperatorModeState = async (cardURL, format = 'isolated') => {
+      let operatorModeStateService = this.owner.lookup(
+        'service:operator-mode-state-service',
+      ) as OperatorModeStateService;
+
+      await operatorModeStateService.restore({
+        stacks: [[{ id: cardURL, format }]],
+      });
+    };
+  });
+
+  test('render a CardDef field (singular) linked to from FieldDef (edit mode)', async function (assert) {
+    let { field, contains, linksTo, CardDef, FieldDef, Component } = cardApi;
+    let { default: StringField } = string;
+    let { default: NumberField } = number;
+
+    class CurrencyCard extends CardDef {
+      static displayName = 'Currency';
+      @field denomination = contains(StringField);
+      @field currencyName = contains(StringField);
+      @field icon = contains(StringField);
+
+      static embedded = class Embedded extends Component<typeof this> {
+        <template>
+          <div class='currency' data-test-currency-embedded>
+            <@fields.icon />
+            <@fields.denomination />
+            -
+            <@fields.currencyName />
+          </div>
+          <style>
+            .currency {
+              display: flex;
+              font-weight: bold;
+            }
+          </style>
+        </template>
+      };
+    }
+
+    class TxAmountField extends FieldDef {
+      @field quantity = contains(NumberField);
+      @field denomination = linksTo(CurrencyCard);
+    }
+
+    class TxCard extends CardDef {
+      static displayName = 'Transaction';
+      @field name = contains(StringField);
+      @field transferAmount = contains(TxAmountField);
+    }
+
+    let usdCard = new CurrencyCard({
+      denomination: 'USD',
+      currencyName: 'United States Dollar',
+      icon: '🇺🇸',
+    });
+
+    let txCard = new TxCard({
+      name: 'Transfer',
+      transferAmount: new TxAmountField({
+        quantity: 250,
+        denomination: usdCard,
+      }),
+    });
+
+    await shimModule(`${testRealmURL}currency`, { CurrencyCard }, loader);
+    await shimModule(`${testRealmURL}tx`, { TxCard }, loader);
+
+    let usdCardDoc = await saveCard(usdCard, `${testRealmURL}usd`, loader);
+    let txCardDoc = await saveCard(txCard, `${testRealmURL}Tx/1`, loader);
+
+    await createTestRealm({
+      '.realm.json': `{ "name": "Local Workspace" }`,
+      'usd.json': usdCardDoc,
+      'Tx/1.json': txCardDoc,
+    });
+    await renderComponent(OperatorModeComponent);
+    await setCardInOperatorModeState(`${testRealmURL}Tx/1`, 'edit');
```

Comment:
After only calling `saveCard`, I wasn't directly able to call `setCardInOperatorModeState`. The instances were not found. I changed `saveCard` to return a card doc, and then created a realm with those docs. This worked. It's much simpler to write tests this way as opposed to how we were writing tests in `operator-mode-test` file. We used this shimming-and-saving method in `card-basics-test` file. I adjusted it to add the realm and operator mode info. Because the cards only have a card-header in operator mode.